### DISCUSSION TITLE
How to ensure workspaces are outputted to the right monitors after renaming the workspaces dynamically

### DISCUSSION THREAD

**Discussion by jtsao22** [i3/i3#5255] [Q&A]
Hello,

First off, I love i3 and thanks so much for the work. I've been using it for several years, but never quite found the right way to do this.

I use the script below to rename my workspace (and bindsym'd it to a keyboard shortcut).

```bash
# For regolith, which adds spans to the workspaces, use this:
OLD=$(i3-msg -t 'get_workspaces' | sed -e 's/num/\n{\"num/g' | grep \"focused\":true | sed -e 's/,"/\n/g' | grep name | cut -d\" -f 3)
i3-input -F "rename workspace \"$OLD\" to %s" -P 'New Name: '
```
 It works, but whenever I remove and then reconnect my monitor from my laptop, the workspaces no longer go to the right monitors. This is probably because my workspaces are set with static variables.

```bash
workspace "$ws1" output HDMI-1
workspace "$ws2" output HDMI-1
workspace "$ws3" output HDMI-1
workspace "$ws4" output HDMI-1
workspace "$ws5" output HDMI-1
workspace "$ws6" output eDP-1
workspace "$ws7" output eDP-1
workspace "$ws8" output eDP-1
workspace "$ws9" output eDP-1
workspace "$ws10" output eDP-1
```
I tried setting the ws* values from the i3 script, but that didn't seem to work. 

Is there some way to set a workspace **number** to an output? Or is there a better way to do this?

Thank you!

**Comment by benibilme** [comment]
Possible not the solution, I would like to give an idea. 

I keep different set of configuration files based on the labtop, desktop connections. I use ansible to configure the config files based host environment. I split the common and platform specific files. In the config file, I arrange the inclusions based on the platform. You need an external scripting because i3 does not give you any command to configure config file at run time. 

my ansible i3 role file structure is as follows which may give an idea. optiplex is my desktop, vostro is my labtop

````
├── files
│   ├── common
│   │   ├── config-bar
│   │   ├── config-binds
│   │   ├── config-exec
│   │   ├── config-modes
│   │   ├── config-movements
│   │   ├── config-scratchpad
│   │   └── i3exit
│   └── platform
│       ├── optiplex
│       │   ├── config
│       │   ├── config-device
│       │   ├── config-workspaces
│       │   └── i3status.conf
│       └── vostro
│           ├── config
│           ├── config-device
│           ├── config-kde
│           ├── config-workspaces
│           └── i3status.conf
└── tasks
    └── main.yml
````

My config file for optiplex becomes after playing ansible-playbook as follows 

````
font pango:monospace 12

set $mod Mod4
set $term alacritty
set $up m
set $down k
set $left t
set $right l

floating_modifier $mod

include config-binds
include config-movements
include config-workspaces
include config-modes
include config-bar
include config-scratchpad
include config-exec
include config-device
````

Ansible creates .i3 directory and copy correct files in a flat directory structure based on the hostname. Keeping different kind of configuration for each platform will give flexibility for other things as well.  I can keep hard coded display names in the config-workspace different for each platform.

**Comment by jtsao22** [comment]
Thanks @benibilme , I like the config. I use stow for a similar effect, but mine definitely needs some cleanup.

Anyway, unfortunately it's not what I'm looking for. My questions is rather specific to dynamically changing the workspace names and making sure they are still displayed on the right monitors after unplugging/replugging or coming back from suspend.

**Reply by ST92** [reply]
I'd say it would call for workspaces to have additional names as aliases, ones that don't change. I'd put it up as a feature request.
